### DISCUSSION TITLE
Where are the panel temporary files on MacOs?

### DISCUSSION THREAD

**Discussion by dobo365** [RomanKubiak/ctrlr#188] [Q&A]
Thx  in advance!

**Comment by Godlike-Productions** [comment]
Open and run the Demo - Files panel.  The drop down at the bottom gives you the various paths on your computer.  I'm on windows, and a few of them are wonky, but if you run them on a Mac, you should be able to work it out.

**Reply by dobo365** [reply]
Sorry it doesn't work. It indeed gives very strange results ;-)
As you know, on Windows they are in c:/users/your_user/appdata/roaming/ctrlr then with a folder corresponding to the panel ID
I'm also on PC but using a Mac in a VM.
I don't find the temp folder as I'm not a frequent Mac user ;-)

**Reply by Godlike-Productions** [reply]
I'm not a mac user either, but I'll try to remember to look for this when I build my current panels on the mac.

**Comment by dnaldoog** [comment]
I use a VirtualBox Mojave to build VSTs and Standalones for Ctrlr.

You can see the path at the bottom of  'finder'  - Next to controller is the folder for the 'Juno Alpha' standalone version and I think this folder is both shared by component files and vst files.

so something like <code>/Users/Goodweather/Library/Preferences/yourapp/</code> perhaps?

![macpathtoctrlr](https://user-images.githubusercontent.com/76831683/109232600-e660a980-780b-11eb-9c74-cd0ab86fe469.png)

Personally, I have never used MacOS so don't know an awful lot about it! ;)

**Comment by dobo365** [comment]
OK guys!
I knew in fact it was in Library/Preferences but never found them. Performing a Search didn't find any Ctrlr stuff either.
Here is the explanation:
- the temporary files are indeed in Library/Preferences BUT not at the computer level. There are - as indicated by @dnaldoog here above - at the user level
- BUT they are hidden as the AppData folder is hidden on Windows and this is the reason I didn't see anything
- To reach the hidden folder or unhide this personal Library folder, use the Finder and press the Alt (Option button) key when clicking on Go. The Go menu will show the Library entry. Another possibility is to use Goto Folder in the Go menu and to type ~/Library

By this, everybody knows where the Ctrlr temp files are located both on PC and Mac.
Happy coding :-)
